src/bp/ui-studio/src/web/views/FlowBuilder/sidePanel/flows/index.jsx

```diff
@@ -90,6 +101,25 @@ export default class FlowsList extends Component {
       }
     })
 
+    const handleRename = handleAction(() => {
+
+      const name = window.prompt('Please enter the new name for that flow', flow.name.replace(/\.flow\.json$/i, ''))
+  
+      if (!name) {
+        return
+      }
+  
+      if (/[^A-Z0-9-_\/]/i.test(name)) {
+        return alert('ERROR: The flow name can only contain letters, numbers, underscores and hyphens.')
+      }
+  
+      if (name !== flow.name && includes(this.props.flows.map(f => f.name), name + '.flow.json')) {
+          return alert(`ERROR: The flow ${name} already exists`)
+      }
+      
+      this.props.renameFlow({targetFlow: flow.name, name : `${name}.flow.json`})
+    })
```

Comment:
I know you just moved the code, but it would make much more sense in its own modal component :_P

Reply:
right, in a next PR 🚀